lib/os/mpsc_pbuf.c

```diff
@@ -5,6 +5,26 @@
  */
 #include <sys/mpsc_pbuf.h>
 
+#define MPSC_PBUF_DEBUG 0
+
+#define MPSC_PBUF_DBG(buffer, ...) do { \
+	if (MPSC_PBUF_DEBUG) { \
+		printk(__VA_ARGS__); \
+		if (buffer) { \
+			mpsc_state_print(buffer); \
+		} \
+	} \
+} while (0)
+
+static inline void mpsc_state_print(struct mpsc_pbuf_buffer *buffer)
+{
+	if (MPSC_PBUF_DEBUG) {
+		printk("wr:%d/%d, rd:%d/%d\n",
+			buffer->wr_idx, buffer->tmp_wr_idx,
+			buffer->rd_idx, buffer->tmp_rd_idx);
+	}
+}
```

Comment:
Do these require `CONFIG_LOG_PRINTK` to be disabled?

Reply:
no, when `CONFIG_LOG_PRINTK` is enabled then this module is not used by the logger so it can printk without issue. I've added it for debugging purposes during testing. Since default logging relies on it, i didn't want to use the logger.